File: src/FCUpgrade.API/test_nuxt.py
import urllib.request
import re
import json

req = urllib.request.Request('https://en.fifaaddict.com/fo4db?page=1', headers={'User-Agent': 'Mozilla/5.0'})
html = urllib.request.urlopen(req).read().decode('utf-8')
m = re.search(r'window\.__NUXT__=(.+?);[\s]*</script>', html, re.DOTALL)
if m:
    js_code = m.group(1)
    # the js_code might not be valid JSON (it has unquoted keys, functions, etc)
    # Let's just do a regex count of how many items there are, or write a node.js script instead which can execute it!
    print("Found NUXT")

File: src/FCUpgrade.API/test_nuxt.js
const https = require('https');

https.get('https://en.fifaaddict.com/fo4db?page=1', (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => {
        const match = data.match(/window\.__NUXT__=(.+?);[\s]*<\/script>/s);
        if (match) {
            const jsCode = match[1];
            // Evaluate in a safe-ish way
            const fn = new Function('window', `window.__NUXT__ = ${jsCode}; return window.__NUXT__;`);
            const nuxt = fn({});
            if (nuxt.data && nuxt.data[0] && nuxt.data[0].items) {
                const items = nuxt.data[0].items;
                console.log(`Page 1 items count: ${items.length}`);
                console.log(`First item: ${items[0].name}`);
                console.log(`Season: ${items[0].season_name}`);
            } else {
                console.log("No items found in data[0]");
            }
        }
    });
});
